transports/sftp/src/main/java/org/mule/transport/sftp/SftpReceiverRequesterUtil.java

```diff
@@ -325,78 +337,89 @@ private InputStream archiveFile(SftpInputStream is, File archiveFile) throws IOE
     }
 
     /**
-     * Checks if the file has been changed.
-     * <p/>
-     * Note! This assumes that the time on both servers are synchronized!
-     * 
-     * @param fileName The file to check
-     * @param client instance of StftClient
+     * Evaluate if a file meets the conditions to be processed.
+     *
+     * @param fileName The name of the file to evaluate
+     * @param client instance of SftpClient
+     * @param checkFileAge indicates if the fileAge condition should be evaluated.
      * @param fileAge How old the file should be to be considered "old" and not
      *            changed
-     * @param sizeCheckDelayMs Wait time (in ms) between size-checks to determine if
-     *            a file is ready to be processed.
-     * @return true if the file has changed
+     * @return true if the file meets the conditions to be processed.
      * @throws Exception Error
      */
-    protected boolean canProcessFile(String fileName, SftpClient client, long fileAge, long sizeCheckDelayMs)
-        throws Exception
+    protected boolean canProcessFile(String fileName, SftpClient client, boolean checkFileAge, long fileAge)
```

Comment:
It makes no sense to pass a checkFileAge=false and a fileAge, so only call this method when it is required and remove the checkFileAge parameter